components/lifesensor_macro/include/macro_task.h

```diff
@@ -0,0 +1,72 @@
+#ifndef __MACRO_TASK__
+#define __MACRO_TASK__
+
+#include <freertos/task.h>
+
+
+/**
+ * struct lifesensor_task_t - generic task struct
+ * @tcb:    Holds task related (internal) data.
+ * @handle: Provides a reference to the task.
+ * @stack:  Stores the address of the tasks' stack.
+ *
+ * This struct serves as a shortcut for operating with FreeRTOS tasks such
+ * that it includes necessary fields for managing a tasks data. It is intended
+ * to be used together with the LIFESENSOR_TASK() macro.
+ */
+typedef struct {
+	StaticTask_t tcb;
+	TaskHandle_t handle;
+	StackType_t stack[];
+} lifesensor_task_t;
+
+
+/**
+ * LIFESENSOR_TASK() - Generic task struct declaration macro
+ * @stacksize:  Determines the stack depth used.
+ *
+ * The macro is used for easier task struct initialization. The resulting
+ * task struct should be initialized through the corresponding 
+ * `lifesensor_task_init()` function.
+ *
+ * NOTE:	`task` houses the actual task struct of type `lifesensor_task_t`.
+ * 			`lifesensor_task_t`s stack attribute of undefined length will
+ * 			point to the stack declared here with size `stacksize`.
+ *
+ */
+#define LIFESENSOR_TASK(stacksize)                                          \
+	struct {                                                                \
+		lifesensor_task_t task;                                             \
+		StackType_t stack[stacksize];                                       \
+	}
+
+
+/**
+ * lifesensor_task_init() - Initialize a task generated with LIFESENSOR_TASK
+ * @anon_task:	Pass the struct created with LIFESENSOR_TASK.
+ * @name:		Name the task to recognize it later.
+ * @func:		Provide a function pointer to execute in task.
+ * @args:		Function arguments that will be fed to `func`.
+ * @prio:		Set the desired task priority.
+ *
+ * This is a wrapper for `xTaskCreateStatic()` which initializes several
+ * internal variables. It provides another layer of abstraction to not bother
+ * much with the FreeRTOS internals and hides redundant code.
+ *
+ * Please consult https://www.freertos.org/xTaskCreateStatic.html
+ */
+
+#define lifesensor_task_init(anon_task, name, func, args, prio)             \
+{                                                                           \
+	anon_task.task.handle = xTaskCreateStatic(                              \
+		func,                                                               \
+		name,                                                               \
+		sizeof(anon_task.stack) / sizeof(anon_task.stack[0]),               \
+		args,                                                               \
+		prio,                                                               \
+		anon_task.stack,                                                    \
+		(StaticTask_t*)(&anon_task.task.tcb)                                \
+	);                                                                      \
```

Comment:
```suggestion
	(*task).task.handle = xTaskCreateStatic(                              \
		func,                                                               \
		name,                                                               \
		sizeof((*task).stack) / sizeof((*task).stack[0]),               \
		args,                                                               \
		prio,                                                               \
		(*task).stack,                                                    \
		(StaticTask_t*)(&(*task).task.tcb)                                \
	);                                                                      \
```

Reply:
Would fail, too much text would be replaced.